mass/adapters/outbound/dao.py

```diff
@@ -64,3 +67,27 @@ def get_dao(self, *, class_name: str) -> ResourceDao:
             return self._resource_daos[class_name]
         except KeyError as err:
             raise DaoNotFoundError(class_name=class_name) from err
+
+    def collection_init_and_index_creation(self) -> None:
+        # get client
+        client: MongoClient = MongoClient(
+            self._config.db_connection_str.get_secret_value()
+        )
+        db = client[self._config.db_name]
```

Comment:
Also, maybe better to create the client in the init method, similar to how MongoDbDaoFactory does it.

In that case you shouldn't close the client of course.

Reply:
You should also remove the manual creation of the index in the fixture now.

Reply:
> Also, maybe better to create the client in the init method, similar to how MongoDbDaoFactory does it.
> 
> In that case you shouldn't close the client of course.

I think that would make sense if we were going to use the client more than once or in another method, but it's only used once and inside this method.

Reply:
- The boolean was added in [e8b9e8d](https://github.com/ghga-de/mass/pull/13/commits/e8b9e8df779ccd22b734db53f296f943bec92bf8)
- Closed the client in [d17935e](https://github.com/ghga-de/mass/pull/13/commits/d17935ecccd32831b3a101c58c180fc643140cec)
- Changed the joint fixture's load data function in [efb7bae](https://github.com/ghga-de/mass/pull/13/commits/efb7baecbcc5cd2d39dec51f330f6c3f34a52774) to use load_resource() instead of interacting with the client directly